cpp/src/parquet/column_writer.h

```diff
@@ -53,6 +54,8 @@ class Encryptor;
 class OffsetIndexBuilder;
 class WriterProperties;
 
+using KeyValueMetadata = ::arrow::KeyValueMetadata;
```

Comment:
It's generally better to avoid `using` directives in non-internal `.h` files.

Reply:
We have to choose either this or include "parquet/metadata.h" for consistent use of this class

Reply:
IMO just `::arrow::KeyValueMetadata` is ok here